File: src/SkyBlock/enchants/armor/Gears.php
<?php


namespace SkyBlock\enchants\armor;


use pocketmine\entity\effect\EffectInstance;
use pocketmine\entity\effect\VanillaEffects;
use pocketmine\player\Player;

class Gears extends BaseArmorEnchant {

    static int $id = 131;

    public function isApplicableTo(Player $holder) : bool {
        return true;
    }

    /**
     * @param Player $attacker
     * @param Player $victim
     * @param int    $enchantmentLevel
     */
    public function onActivation(Player $attacker, Player $victim, int $enchantmentLevel) : void {
        if (!$victim->getEffects()->has(VanillaEffects::SPEED())) {
            $effect = new EffectInstance(VanillaEffects::SPEED(), $enchantmentLevel * 2 * 20, mt_rand(1, ceil($enchantmentLevel / 4)), false);
            $victim->getEffects()->add($effect);
        }
    }

}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/armor/Gears.java
package dev.skyblock.enchants.armor;

import dev.skyblock.enchants.BaseArmorEnchant;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;
import java.util.concurrent.ThreadLocalRandom;

public class Gears extends BaseArmorEnchant {
    public static final int ID = 131;

    public Gears() {
        super(ID, "Gears", 15);
    }

    @Override
    public boolean isApplicableTo(Player player, int level) {
        return true;
    }

    @Override
    public void onActivation(Player victim, EntityDamageByEntityEvent ev, int level) {
        if (!victim.hasPotionEffect(PotionEffectType.SPEED)) {
            int amplifier = ThreadLocalRandom.current().nextInt(0, (int) Math.ceil(level / 4.0));
            victim.addPotionEffect(new PotionEffect(PotionEffectType.SPEED, level * 2 * 20, amplifier));
        }
    }
}
